src/openforms/submissions/templates/report/submission_report.html

```diff
@@ -63,7 +67,14 @@ <h2 class="subtitle">{{ submission_step_node.render }}</h2>
                                             </div>
 
                                             <div class="submission-step-row__value{% if not component_node.display_value %} submission-step-row__value--empty{% endif %}">
-                                                {{ component_node.display_value|default:"" }}
+                                                {% if not component_node.display_value %}
+                                                {% elif component_node.component.type == "textfield" and component_node.component.multiple %}
+                                                    {% include 'report/submission_report_node_list.html' with values=component_node.value_list %}
+                                                {% elif component_node.component.type == "selectboxes" or component_node.component.type == "select" %}
+                                                    {% include 'report/submission_report_node_list.html' with values=component_node.value_list %}
+                                                {% else %}
+                                                    {{ component_node.display_value|default:"" }}
+                                                {% endif %}
```

Comment:
hardcoding component types in these templates breaks the entire concept of a renderer/tree visitor where each node type controls its own behaviour. It spreads implementation details across multiple places and tracking down which code results in which display becomes much harder.

All this means that the `display_value` attribute is responsible for outputting the correct markup given the current context (render mode, html on/off). So, these template changes should be reverted IMO in favour of another approach.

Reply:
I've removed the whole `value_list` concept, so i've reverted this change.

It felt weird to add html on any other level than the html file and I thought only about the use case surrounding the submission report, but i understand your point. It did create a weird situation and, with the second implementation, i've found some other places that benefit from the html list view